### PR TITLE
Update timeout configuration to match spec

### LINKED ISSUE #131: Potentially incorrect timeout behavior
According to the [Desktop Notifications Specification](https://specifications.freedesktop.org/notification-spec/notification-spec-latest.html), `expire_timeout` is:

>  The timeout time in milliseconds since the display of the notification at which the notification should automatically close.
>
> If -1, the notification's expiration time is dependent on the notification server's settings, and may vary for the type of notification. If 0, never expire. 

But in [my calendar notification tool](https://git.sr.ht/~bsprague/dont-miss-it), I've set [the timeout to never expire](https://git.sr.ht/~bsprague/dont-miss-it/tree/main/item/notifier.go#L97) ([which does indeed have a value of `0`](https://github.com/esiqveland/notify/blob/120b9022f591d4c502516e691acec335960d406d/notification.go#L63)), and I noticed it was still getting the default `timeout` specified in my `wired.ron` config.

I believe that's happening in [this code here](https://github.com/Toqozz/wired-notify/blob/d65f227779061299842b4e1891954c4843ee2750/src/bus/dbus.rs#L455-L458), which should read `if timeout < 0 { ... }` instead, and `timeout == 0` should be handled specially.

If I've gotten that correct, I'm happy to attempt to take this on (with a few pointers of how to do it)

### PR DESCRIPTION
**Pull request by bcspragu**
Fixes #131 by changing behavior to:

- Use default for timeouts less than zero
- Never expire for timeouts equal to zero
- Use the timeout otherwise

I built and tested this locally (using `notify-send` and others), it works as expected

Also adds a new `timeout_behavior = {Legacy, DBusSpec}` config option, which defaults to the new behavior. The wiki should probably be updated after this lands as well.

### CODE REVIEW

**Comment by Toqozz** [comment]
Sorry, for some reason I thought `timeout` was a float, but it's actually a `u32` in milliseconds.

I do think forever should actually mean forever, not just a big number, so I think we should change `timeout` to be an enum:

```rust
enum Timeout {
    Forever,
    Milliseconds(u32),
}
```

And then use that appropriately, similar to how you've done already.  `dbus.rs` is probably a fine place for that.

In `window.rs` you'll probably need to make `fuse` a `Timeout` as well, and update it accordingly.  I'm not sure what all that will look like yet, your choice! 

Also, we do want to have a new config option for the new behaviour, so users can go back to the old if they want.  I think I'm OK with it being a breaking change, though.

Please let me know if you have any questions or thoughts!

**Comment by bcspragu** [comment]
That all sounds good, I'll give it a shot. Moving this to a draft in the mean time

**Comment by bcspragu** [comment]
All of the stuff from above should be done, let me know what you think!

File: src/rendering/window.rs [hunk 1]
```diff
@@ -327,12 +327,15 @@ impl NotifyWindow {
...
         if self.update_mode.contains(UpdateModes::FUSE) {
-            self.fuse -= delta_time.as_millis() as i32;
-            if self.fuse <= 0 {
-                // Window will be destroyed after others have been repositioned to replace it.
-                // We can return early because drawing will be discarded anyway.
-                self.marked_for_destroy = true;
-                return true;
+            if let Timeout::Milliseconds(fuse) = self.fuse {
+                let new_fuse = fuse - delta_time.as_millis() as i32;
+                self.fuse = Timeout::Milliseconds(new_fuse);
```
Comment (Toqozz) [inline src/rendering/window.rs:332]: Not sure if this compiles, but something like it should:
```suggestion
            if let Timeout::Milliseconds(ref mut fuse) = self.fuse {
                *fuse -= delta_time.as_millis();
```

**Comment by Toqozz** [comment]
Looks good!  Also be sure to add the option to `config.ron` as well.

File: src/rendering/window.rs (hunk 1 above)
Comment (bcspragu) [reply src/rendering/window.rs]: Good call, done!

**Comment by bcspragu** [comment]
> Looks good! Also be sure to add the option to `config.ron` as well.

Updated, with a bit of documentation.

**Comment by Toqozz** [comment]
Great, thank you for the pull request!

**Pull request merged**